coalib/bearlib/languages/documentation/DocBaseClass.py

```diff
@@ -254,31 +256,72 @@ def extract_documentation_with_markers(content, docstyle_definition):
             yield doc
 
 
-def extract_documentation(content, language, docstyle):
+class DocBaseClass:
     """
-    Extracts all documentation texts inside the given source-code-string using
-    the coala docstyle definition files.
-
-    The documentation texts are sorted by their order appearing in ``content``.
-
-    For more information about how documentation comments are identified and
-    extracted, see DocstyleDefinition.doctypes enumeration.
-
-    :param content:            The source-code-string where to extract
-                               documentation from. Needs to be a list or tuple
-                               where each string item is a single line
-                               (including ending whitespaces like ``\\n``).
-    :param language:           The programming language used.
-    :param docstyle:           The documentation style/tool used
-                               (e.g. doxygen).
-    :raises FileNotFoundError: Raised when the docstyle definition file was not
-                               found.
-    :raises KeyError:          Raised when the given language is not defined in
-                               given docstyle.
-    :raises ValueError:        Raised when a docstyle definition setting has an
-                               invalid format.
-    :return:                   An iterator returning each DocumentationComment
-                               found in the content.
+    DocBaseClass holds important function which will extract, parse
+    and generates diffs for documentation.
```

Comment:
You can expand a bit on this docstring, for example add "All bears that process documentation should inherit from this" or something.